### ISSUE TITLE
Possible memory leak(ish) part in ImageCacheRenderer

### ISSUE THREAD

**Issue by mzimdars-service1-com** [molinch/FFImageLoading#1509]
## 🐛 Bug Report

Hi first of all a big **thank you** for your work, it is helping us a lot!

I am currently hunting memory leaks in our app and found one suspicious code line in CachedImageRenderer
in 'UpdateImage' an 'IScheduledWork' is created that holds a reference to the target view. this task is stored in '_currentTask'
when the scheduled work is done, the _currentTask is not cleared, thus still has an (indirect) reference to the target view.

The target view references the complete view its in (through 'parent' relationship)

As long as the image is cached (what we want), the view is kept alive, at least until the image is used elsewhere (and the renderer is re-used), as then a new _currentTask is set

### Expected behavior

When the scheduled work is done, the _currentTask should be cleared (as the last action of the work) or the _currentTask should be a weak reference (should be enough for cancelling the task as long as it is pending in the systems dispatcher)

### Reproduction steps
It should be enough to use a cached image renderer within a view, close it and the garbage collector then should be able to collect the view (which is not happening due to the reference through _currentTask)
### Configuration

**Version:** 2.4.11.982 (Xamarin.FFImageLoading.Svg.Forms)

**Platform:** 
- [ ] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [X] :earth_americas: UWP (profiled platform, issue could be in other platforms as well)
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [X] :monkey: Xamarin.Forms

**Comment by DevEddy** [comment]
Same issue here

**Comment by mzimdars-service1-com** [comment]
In fact, we fixed a memory leak issue that we had in our code before christmas. This released controls that kept a reference to the renderer and this this issue occured.
But with our fix, the reference to the renderer disappeared and thus this renderer and the _currentTask was freed as well.
So from our side, this issue can be closed, because it is not-a-bug

@DevEddy most probably your memory leak is somewhere else and just keeps the renderer alive, but I can say, this is not the cause.

We often have memory leak issues with "static" (real static or long-lved central services) event handler that were subscribed by (view) models used in Xamarin Forms controls, where the event handlers are not unsubscribed when the control/(view) model is not longer needed. This construct keepos the complete control and (view) model alive, and with it the associated renderers.
Good luck finding the real cause!
mz

**Comment by DevEddy** [comment]
Hi @mzimdars-gms-online-de ,

thank you for the response. And damn you were right... 
For me it was the `HttpClientHandler `with the `ServerCertificateCustomValidationCallback `event.
When passing the handler to the `HttpClient `I set the parameter `disposeHandler `to true. 

I saw in the diagnostic tool of Visual Studio that there is a strong reference to the `TypedHandler` callback, the reference count increased every time I used the `HttpClient`. But the weird thing is that it only affected the `HttpClient` from the configuration in `FFImageLoading`.

I am reusing the `HttpClientHandler `now and the memory gets freed as expected.

-- Eddy
